### PR TITLE
Allow send `SSH into runner` info. to DM

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Easier to get notified.

### CODE REVIEW

File: .github/workflows/ssh-runner.yml [hunk 1]
```diff
@@ -53,11 +53,21 @@ jobs:
       - name: NVIDIA-SMI
         run: |
           nvidia-smi
-      
+
+      - name: Store Slack infos
+        #because the SSH can be enabled dynamically if the workflow failed, so we need to store slack infos to be able to retrieve them during the waitforssh step
+        shell: bash
+        run: |
+          if [ "${{ secrets[format('{0}_{1}', github.actor, 'SLACK_ID')] }}" != "" ]; then
```
Comment (ydshieh) [inline .github/workflows/ssh-runner.yml:61]: If a GitHub secret is added to the repository: for example, `YDSHIEH_SLACK_ID` with a value being the corresponding slack user ID

Comment (ArthurZucker) [reply .github/workflows/ssh-runner.yml:61]: Quick Q, should the slack id not be an argument to  workflow_dispatch 🤗

Comment (ydshieh) [reply .github/workflows/ssh-runner.yml:61]: It is an ID like `X13JMGOCTSU`, and has to be copied from the Slack user profile.
Being an input might add some friction to this process (but browser usually remember input)

The IMPORTANT is: we want to keep it as a secret (and not show on log), like what we did so for 

> ${{ secrets.SLACK_CIFEEDBACK_CHANNEL }}

File: .github/workflows/ssh-runner.yml [hunk 2]
```diff
@@ -53,11 +53,21 @@ jobs:
...
+            echo "SLACKCHANNEL=${{ secrets[format('{0}_{1}', github.actor, 'SLACK_ID')] }}" >> $GITHUB_ENV
+          else
+            echo "SLACKCHANNEL=${{ secrets.SLACK_CIFEEDBACK_CHANNEL }}" >> $GITHUB_ENV
```
Comment (amyeroberts) [inline .github/workflows/ssh-runner.yml:64]: As we're adding this to the GITHUB_ENV, is this less secure? Previously the information was stored in secrets - will making it persistent like this make it more accessible?

Comment (ydshieh) [reply .github/workflows/ssh-runner.yml:64]: No, it would still be masked by GitHub Actions.

At least, it follows what has been done by Guillaume.

**Comment by ydshieh** [comment]
@ArthurZucker  @LysandreJik

**Review by LysandreJik — APPROVED** [review]
Cool ! Can you have @glegendre01 review this just for security purposes?

**Comment by ydshieh** [comment]
Thanks for the review. Sure I will ping him.

File: .github/workflows/ssh-runner.yml (hunk 2 above)
Comment (glegendre01) [reply .github/workflows/ssh-runner.yml:64]: we need to test. normally github keep it obfuscated on the logs.
if it's not the case, eventually we can add this kind of command : `echo "::add-mask::$the_secret"
`

Comment (glegendre01) [reply .github/workflows/ssh-runner.yml:64]: let's try this PR

**Review by glegendre01 — APPROVED** [review]

File: .github/workflows/ssh-runner.yml (hunk 2 above)
Comment (ydshieh) [reply .github/workflows/ssh-runner.yml:64]: FYI: there is a run page

https://github.com/huggingface/transformers/actions/runs/10736292447/job/29775922690

In its raw log page, searching `SLACKCHANNEL`, all places are masked `***`

**Pull request merged**